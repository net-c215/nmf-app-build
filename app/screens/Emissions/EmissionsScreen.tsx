import React from "react";
import { View } from "react-native";
import { useSelector } from "react-redux";
import { useNavigation } from "@react-navigation/native";

import { Text, Button } from "components";
import { t } from "utils";
import { navigate } from "navigation";

import styles from "./EmissionsScreen.styles";
import { EmissionsList } from "./components";
import { selectors } from "./ducks";

const EmissionsScreen = () => {
  const emissions = useSelector(selectors.getEmissions);
  const navigation = useNavigation();
  const navigator = navigate(navigation);

  return (
    <>
      <View style={styles.container}>
        <EmissionsList emissions={emissions} />
      </View>
      <View style={styles.buttonView}>
        <Button.Primary
          fullWidth
          onPress={() => navigator.openCategorySelection()}
          textType={"Secondary"}
        >
          <Text.Secondary numberOfLines={1} center white bold>
            {t("EMISSIONS_SCREEN_ADD_EMISSION")}
          </Text.Secondary>
        </Button.Primary>
      </View>
    </>
  );
};

export default EmissionsScreen;
