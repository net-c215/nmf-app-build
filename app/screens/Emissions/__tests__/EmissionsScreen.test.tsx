import React from "react";
import renderer from "react-test-renderer";
import EmissionsScreen from "../EmissionsScreen";

const props = {
  navigation: {
    push: () => {
      // do nothing.
    }
  }
};

it("EmissionsScreen renders correctly", () => {
  const tree = renderer.create(<EmissionsScreen {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
