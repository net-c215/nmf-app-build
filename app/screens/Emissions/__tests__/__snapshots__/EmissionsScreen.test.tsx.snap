// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmissionsScreen renders correctly 1`] = `
Array [
  <RCTScrollView
    style={
      Object {
        "flex": 1,
        "paddingTop": 10,
      }
    }
  >
    <View>
      <TabbedView
        items={
          Array [
            Object {
              "component": <View
                style={
                  Object {
                    "flex": 1,
                    "paddingTop": 20,
                  }
                }
              >
                <EmissionListItem
                  restaurant={true}
                  subTitle="2.1 kg CO2"
                  title="170 g of red meat"
                />
                <EmissionListItem
                  build={true}
                  subTitle="2.1 kg CO2"
                  title="170 g of red meat"
                />
                <EmissionListItem
                  plane={true}
                  subTitle="2.1 kg CO2"
                  title="170 g of red meat"
                />
              </View>,
              "title": "To offset",
            },
            Object {
              "component": <View
                style={
                  Object {
                    "flex": 1,
                    "paddingTop": 20,
                  }
                }
              >
                <EmissionListItem
                  build={true}
                  subTitle="2.1 kg CO2"
                  title="170 g of red meat"
                />
                <EmissionListItem
                  plane={true}
                  subTitle="2.1 kg CO2"
                  title="170 g of red meat"
                />
              </View>,
              "title": "Mitigated",
            },
          ]
        }
      />
    </View>
  </RCTScrollView>,
  <View
    style={
      Object {
        "bottom": 0,
        "flexDirection": "row",
        "left": 0,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <Button.Primary
      black={true}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
          "marginBottom": 20,
          "marginLeft": 14,
          "marginRight": 7,
        }
      }
      textType="Primary"
    >
      <Text.Primary
        bold={true}
        center={true}
        white={true}
      >
        Mitigate
      </Text.Primary>
    </Button.Primary>
    <Button.Primary
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
          "marginBottom": 20,
          "marginLeft": 7,
          "marginRight": 14,
        }
      }
      textType="Primary"
    >
      <Text.Primary
        bold={true}
        center={true}
        white={true}
      >
        Add
      </Text.Primary>
    </Button.Primary>
  </View>,
]
`;
